pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -0,0 +1,183 @@
+}
+
+//Store interface is to allow Hint Providers to retrieve pod affinity
+type Store interface {
+	GetAffinity(podUID string, containerName string) TopologyHints
+}
+
+type containers map[string]TopologyHints
+
+var _ Manager = &manager{}
+
+type policyName string
+
+//NewManager creates a new TopologyManager based on provided policy
+func NewManager(topologyPolicyName string) Manager {
+	klog.Infof("[topologymanager] Creating topology manager with %s policy", topologyPolicyName)
+	var policy Policy
+
+	switch policyName(topologyPolicyName) {
+
+	case PolicyPreferred:
+		policy = NewPreferredPolicy()
+
+	case PolicyStrict:
+		policy = NewStrictPolicy()
+
+	default:
+		klog.Errorf("[topologymanager] Unknow policy %s, using default policy %s", topologyPolicyName, PolicyPreferred)
+		policy = NewPreferredPolicy()
+	}
+
+	var hp []HintProvider
+	pnh := make(map[string]containers)
+	pm := make(map[string]string)
+	manager := &manager{
+		hintProviders:    hp,
+		podTopologyHints: pnh,
+		podMap:           pm,
+		policy:           policy,
+	}
+
+	return manager
+}
+
+func (m *manager) GetAffinity(podUID string, containerName string) TopologyHints {
+	return m.podTopologyHints[podUID][containerName]
+}
+
+func (m *manager) calculateTopologyAffinity(pod v1.Pod, container v1.Container) TopologyHints {
+	socketMask := socketmask.NewSocketMask(nil)
+	var maskHolder []string
+	count := 0
+	affinity := true
+	for _, hp := range m.hintProviders {
+		topologyHints := hp.GetTopologyHints(pod, container)
+		if topologyHints.Affinity && topologyHints.SocketAffinity != nil {
+			klog.Infof("[topologymanager] Topology Affinity.")
+			socketMask, maskHolder = socketMask.GetSocketMask(topologyHints.SocketAffinity, maskHolder, count)
+			count++
+		} else if topologyHints.Affinity && topologyHints.SocketAffinity == nil {
+			klog.Infof("[topologymanager] NO Topology Affinity.")
+			return TopologyHints{
+				SocketAffinity: []socketmask.SocketMask{socketMask},
+				Affinity:       false,
+			}
+		} else if !topologyHints.Affinity && topologyHints.SocketAffinity != nil {
+			klog.Infof("[topologymanager] Cross Socket Topology Affinity")
+			affinity = false
+			socketMask, maskHolder = socketMask.GetSocketMask(topologyHints.SocketAffinity, maskHolder, count)
+			count++
+		}
+	}
+	return TopologyHints{
+		SocketAffinity: []socketmask.SocketMask{socketMask},
+		Affinity:       affinity,
+	}
+}
+
+func (m *manager) AddHintProvider(h HintProvider) {
+	m.hintProviders = append(m.hintProviders, h)
+}
+
+func (m *manager) AddPod(pod *v1.Pod, containerID string) error {
+	m.podMap[containerID] = string(pod.UID)
+	return nil
+}
+
+func (m *manager) RemovePod(containerID string) error {
+	podUIDString := m.podMap[containerID]
+	delete(m.podTopologyHints, podUIDString)
+	delete(m.podMap, containerID)
+	klog.Infof("[topologymanager] RemovePod - Container ID: %v podTopologyHints: %v", containerID, m.podTopologyHints)
+	return nil
+}
```

Comment:
Why call this `RemovePod()` instead of `RemoveContainer()` since it only removes one containers info from the pod?

Also, when it is called in your follow-on PR in the container lifecycle code:
https://github.com/kubernetes/kubernetes/pull/74357/files#diff-81cef540f416b0451c18470420adc42dR71

it sits next to the CPUManagers `RemoveContainer()` call.